### ISSUE TITLE
Empty typeahead throws errors for keyboard events

### ISSUE THREAD

**Issue by jindrichmynarz** [Day8/re-com#195]
When the typeahead component is in focus and empty and the user presses arrow up or down it throws an error "No item NaN in vector of length 0". When tab is pressed, "No protocol method IAssociative.-assoc defined for type string:" is thrown. The tab problem is fixed by #194

This can be reproduced in the [re-com demo](https://re-com.day8.com.au/#/typeahead).

**Comment by itaysabato** [comment]
Any hope of fixing this soon?
Is there a workaround?

**Comment by itaysabato** [comment]
I'm getting a different error on tab BTW:
```
TypeError: suggestion_to_string is null typeahead.cljs:79:8
    display_suggestion typeahead.cljs:79
    external_model_changed typeahead.cljs:163
    ...
```

**Comment by Gregg8** [comment]
@jindrichmynarz, this has been mostly fixed by #196. You also mention the TAB issue which is being handled by #194 but that's not sorted as yet, so I left this open for now.

**Comment by Gregg8** [comment]
#194 now merged
